back/src/Guide/Types/Core.hs

```diff
@@ -316,8 +353,44 @@ changelog ''Category (Past 9, Past 8) []
 deriveSafeCopySorted 8 'base ''Category_v8
 
 instance Aeson.ToJSON Category where
-  toJSON = Aeson.genericToJSON Aeson.defaultOptions {
-    Aeson.fieldLabelModifier = over _head toLower . drop (T.length "category") }
+  toJSON $(fields 'Category) = Aeson.object [
+    "uid"             Aeson..= categoryUid,
+    "title"           Aeson..= categoryTitle,
+    "created"         Aeson..= categoryCreated,
+    "group"           Aeson..= categoryGroup,
+    "status"          Aeson..= categoryStatus,
+    "notes"           Aeson..= categoryNotes,
+    "items"           Aeson..= categoryItems,
+    "itemsDeleted"    Aeson..= categoryItemsDeleted,
+    "enabledSections" Aeson..= categoryEnabledSections
+    ]
+
+instance Aeson.FromJSON Category where
+  parseJSON = Aeson.withObject "Category" $ \o -> do
+    categoryUid             <- o Aeson..: "uid"
+    categoryTitle           <- o Aeson..: "title"
+    categoryCreated         <- o Aeson..: "created"
+    categoryGroup           <- o Aeson..: "group"
+    categoryStatus          <- o Aeson..: "status"
+    categoryNotes           <- o Aeson..: "notes"
+    categoryItems           <- o Aeson..: "items"
+    categoryItemsDeleted    <- o Aeson..: "itemsDeleted"
+    categoryEnabledSections <- o Aeson..: "enabledSections"
+    pure Category{..}
+
+-- | jsonb encode data through JSON AST.
+--
+-- | If you want more speed then use jsonbBytes, it encodes through raw JSON.
+instance ToPostgres Category where
+  toPostgres = Aeson.toJSON >$< HE.jsonb
+
+instance FromPostgres Category where
+  fromPostgres = resultToEither . Aeson.fromJSON <$> HD.jsonb
+
+-- | Unwarp result to category or fail.
+resultToEither :: Aeson.Result Category -> Category
+resultToEither (Aeson.Success category) = category
+resultToEither (Aeson.Error s) = error $ "fromJSON failed with error: " ++ s
```

Comment:
No, `error` is verboten. Apparently we'll have to use `jsonbBytes` after all because it would let us give back the decoding error (as `Left`) without using `error`.